modules/gdscript/editor/gdscript_highlighter.cpp

```diff
@@ -439,10 +442,15 @@ Dictionary GDScriptSyntaxHighlighter::_get_line_syntax_highlighting_impl(int p_l
 			}
 		}
 
-		if (!in_function_name && in_word && !in_keyword) {
+		if (!in_function_name) {
 			if (prev_text == GDScriptTokenizer::get_token_name(GDScriptTokenizer::Token::SIGNAL)) {
 				in_signal_declaration = true;
-			} else {
+			} else if (is_char && prev_text == GDScriptTokenizer::get_token_name(GDScriptTokenizer::Token::FUNC) && str[j - 1] == '(') {
```

Comment:
Checking it that way can cause an out of bounds crash, I think doing a check to avoid it makes the condition less simple and might be less cheap overall